packages/vite/src/node/plugins/dynamicImportVars.ts

```diff
@@ -144,7 +148,9 @@ export async function transformDynamicImport(
 
   let newRawPattern = posix.relative(
     posix.dirname(importer),
-    await toAbsoluteGlob(rawPattern, root, importer, resolve),
+    normalizeGlobPattern(
+      await toAbsoluteGlob(rawPattern, root, importer, resolve),
+    ),
```

Comment:
Given that the problem is that `toAbsoluteGlob` escaping the characters unnecessary, how about not using that function?
Checking the `toAbsoluteGlob` function, most of the code is handling about globs and escaping the paths. I think it would be trimmed down to:
```js
// replace `normalizeGlobPattern` + `toAbsoluteGlob` with
const dir = importer ? dirname(importer) : root
const normalized = rawPattern[0] === '/' ? posix.join(root, rawPattern.slice(1)) : posix.join(dir, rawPattern)

// pass it to `newRawPattern`
let newRawPattern = posix.relative(posix.dirname(importer), normalized)
```

Reply:
OK, I'll fix this.